SearchTDD/Search/IncludeAllHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Linq;
+namespace Search;
+
+public class IncludeAllHandler : ISearchHandler
+{
+    public ISearchHandler? Next { get; set; }
+
+    public IEnumerable<string> Handle(InvertedIndex invertedIndex, IEnumerable<string> query)
+    {
+        var includeAllQueries = query.Where(x => !x.StartsWith("+") && !x.StartsWith("-"));
+        if (!includeAllQueries.Any())
+        {
+            if (Next == null)
+                return invertedIndex.AllNames;
+            return Next.Handle(invertedIndex, query);
+        }
+
+        IEnumerable<string> answer = invertedIndex.AllNames;
+        foreach (var includeAllQuery in includeAllQueries)
+        {
+            try
+            {
+                answer = answer.Intersect(invertedIndex.Database[includeAllQuery.ToUpper()]);
+            }
+            catch (KeyNotFoundException e)
+            {
+                return new List<string>();
+            }
+        }
+
+        if (Next != null)
+            answer = answer.Intersect(Next.Handle(invertedIndex, query));
```

Comment:
there are two ways to `intersect` or `union` a collection. here since you want to assign the result of `Intersect` you can use below code instead: (it is in-place)
```cs
answer.IntersectsWith( ... )
// for union as well:
answer.UnionWith( ... )
```

Reply:
کتابخانه خاصی باید import بشه؟